Description: Fixed new bug with CRC error after reading an over-length palette
Origin: upstream, https://github.com/glennrp/libpng/commit/3939689e7d9d06ee05411210bc8e605adcff294e
Forwarded: not-needed
Author: Glenn Randers-Pehrson <glennrp@users.sourceforge.net>
Last-Update: 2015-11-17
Applied-Upstream: v1.2.54
---
 pngrutil.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/pngrutil.c
+++ b/pngrutil.c
@@ -600,7 +600,7 @@ png_handle_PLTE(png_structp png_ptr, png
    if (png_ptr->color_type == PNG_COLOR_TYPE_PALETTE)
 #endif
    {
-      png_crc_finish(png_ptr, 0);
+      png_crc_finish(png_ptr, (int) length - num * 3);
    }
 #ifndef PNG_READ_OPT_PLTE_SUPPORTED
    else if (png_crc_error(png_ptr))  /* Only if we have a CRC error */
